File: src/main/java/fi/quanfoxes/parser/nodes/ElseIfNode.java
package fi.quanfoxes.parser.nodes;

import fi.quanfoxes.parser.Context;
import fi.quanfoxes.parser.Node;

public class ElseIfNode extends IfNode {
    public ElseIfNode(Context context, Node condition, Node body) {
		super(context, condition, body);
	}

	@Override
    public NodeType getNodeType() {
        return NodeType.ELSE_IF_NODE;
    }
}

File: Vivid/Parser/Nodes/ElseIfNode.cs
public class ElseIfNode : IfNode
{
	public ElseIfNode(Context context, Node condition, Node body, Position? start, Position? end) : base(context, condition, body, start, end)
	{
		Instance = NodeType.ELSE_IF;
	}

	public IfNode GetRoot()
	{
		var iterator = Predecessor;

		while (!iterator!.Is(NodeType.IF))
		{
			iterator = iterator.To<ElseIfNode>().Predecessor;
		}

		return iterator.To<IfNode>();
	}
}